packages/flutter_tools/lib/src/test/flutter_platform.dart

```diff
@@ -521,18 +521,29 @@ class FlutterPlatform extends PlatformPlugin {
           } else {
             globals.printTrace('test $ourTestCount: using observatory uri $detectedUri from pid ${process.pid}');
           }
-          processObservatoryUri = detectedUri;
-          {
-            globals.printTrace('Connecting to service protocol: $processObservatoryUri');
-            final Future<vm_service.VmService> localVmService = connectToVmService(processObservatoryUri,
-              compileExpression: _compileExpressionService);
-            localVmService.then((vm_service.VmService vmservice) {
-              globals.printTrace('Successfully connected to service protocol: $processObservatoryUri');
-            });
-          }
-          gotProcessObservatoryUri.complete();
-          watcher?.handleStartedProcess(
-              ProcessEvent(ourTestCount, process, processObservatoryUri));
+          unawaited(DartDevelopmentService.startDartDevelopmentService(
```

Comment:
Do we need to spin this process up for tests? Does it have an impact on test performance?

Reply:
Yes, because they can be debugged via the service protocol. Tooling like Devtools, VSCode, etc. will start using functionality that is only supported via DDS (e.g., synchronizing isolate resuming by waiting for relevant clients to give their approval to resume) so it's best if we're consistently starting DDS for core use cases.

As for performance impacts, this shouldn't hurt performance of tests in most cases since most tests won't actually be making service requests. There may be a slight increase in latency during test setup, but it should be negligible.